test/test_sync.cpp

```diff
@@ -2986,65 +2986,52 @@ TEST(Sync_UploadDownloadProgress_1)
     TEST_DIR(server_dir);
     TEST_CLIENT_DB(db);
 
-    uint_fast64_t global_snapshot_version;
-
+    std::atomic<uint_fast64_t> downloaded_bytes;
+    std::atomic<uint_fast64_t> downloadable_bytes;
+    std::atomic<uint_fast64_t> uploaded_bytes;
+    std::atomic<uint_fast64_t> uploadable_bytes;
+    std::atomic<uint_fast64_t> snapshot_version;
     {
         int handler_entry = 0;
 
         bool cond_var_signaled = false;
         std::mutex mutex;
         std::condition_variable cond_var;
 
-        std::atomic<uint_fast64_t> downloaded_bytes;
-        std::atomic<uint_fast64_t> downloadable_bytes;
-        std::atomic<uint_fast64_t> uploaded_bytes;
-        std::atomic<uint_fast64_t> uploadable_bytes;
-        std::atomic<uint_fast64_t> progress_version;
-        std::atomic<uint_fast64_t> snapshot_version;
-
         ClientServerFixture fixture(server_dir, test_context);
         fixture.start();
 
         Session session = fixture.make_session(db, "/test");
 
         auto progress_handler = [&](uint_fast64_t downloaded, uint_fast64_t downloadable, uint_fast64_t uploaded,
-                                    uint_fast64_t uploadable, uint_fast64_t progress, uint_fast64_t snapshot) {
+                                    uint_fast64_t uploadable, uint_fast64_t snapshot, double, double) {
             downloaded_bytes = downloaded;
             downloadable_bytes = downloadable;
             uploaded_bytes = uploaded;
             uploadable_bytes = uploadable;
-            progress_version = progress;
             snapshot_version = snapshot;
+            ++handler_entry;
+        };
 
-            if (handler_entry == 0) {
+        std::unique_lock<std::mutex> lock(mutex);
+        session.set_progress_handler(progress_handler);
+        session.set_connection_state_change_listener([&](ConnectionState state, util::Optional<ErrorInfo>) {
```

Comment:
why do we need the connection state to change? is there any progress reported before that we want to skip?

Reply:
this was the part of `m_reliable_download_progress` variable check refactoring. Before it was essentially checking the condition for `progress_version == 0`, and this funcionaly the same as to check that callback wasn't called after we connected but not yet received download message.

So you can see below that the snapshot version is still the same after the first commit. And that is exactly how object-store api was receiving it before refactoring.